### PR TITLE
Fix empty checkboxes bug.

### PR DESCRIPTION
**Pull request by adubovskoy**

### CODE REVIEW

File: modules/anu_lms_assessments/src/Plugin/rest/resource/QuestionRestResource.php [hunk 1]
```diff
@@ -246,6 +246,10 @@ protected function processAnswerToQuestion($answer, int $question_id, Assessment
       /** @var \Drupal\paragraphs\ParagraphInterface[] $options */
       $options = $question->get('field_options')->referencedEntities();
       foreach ($options as $option) {
+        // Setting flag for workaround for preventing re saving paragraph and changing parent_id.
```
Comment (rodrigoaguilera) [inline modules/anu_lms_assessments/src/Plugin/rest/resource/QuestionRestResource.php]: I would rephrase a bit like
```
Prevent paragraph from being saved since that changes the parent_id.

File: modules/anu_lms_assessments/src/Plugin/rest/resource/QuestionRestResource.php [hunk 2]
```diff
@@ -246,6 +246,10 @@ protected function processAnswerToQuestion($answer, int $question_id, Assessment
...
+        // Implementation made as patch for entity_reference_revisions module.
```
Comment (rodrigoaguilera) [inline modules/anu_lms_assessments/src/Plugin/rest/resource/QuestionRestResource.php:250]: I don't understand this part. Does this require a patch to the module?

Comment (rodrigoaguilera) [inline modules/anu_lms_assessments/src/Plugin/rest/resource/QuestionRestResource.php:250]: I don't understand this part. Does this require a patch to the module?

**Review by rodrigoaguilera — CHANGES_REQUESTED** [review]
Is this bug a regression from any recent change?

File: modules/anu_lms_assessments/src/Plugin/rest/resource/QuestionRestResource.php (hunk 2 above)
Comment (adubovskoy) [reply modules/anu_lms_assessments/src/Plugin/rest/resource/QuestionRestResource.php:250]: Yes. (( This patch exists on all projects. In future, we'll need refactor this.

File: modules/anu_lms_assessments/src/Plugin/rest/resource/QuestionRestResource.php (hunk 1 above)
Comment (adubovskoy) [reply modules/anu_lms_assessments/src/Plugin/rest/resource/QuestionRestResource.php]: Tnx! done

**Comment by rodrigoaguilera** [comment]
There is also some php code sniffer complain

:thinking:  I see that `modules/anu_lms_assessments/src/Plugin/rest/resource/AssessmentRestResource.php `
Also used to have a `->dontSave` 

Is it possible we have regression there as well?

**Comment by adubovskoy** [comment]
As I understand, now AssessmentRestResource extends QuestionRestResource and we don't need rewrite it.

File: modules/anu_lms_assessments/src/Plugin/rest/resource/QuestionRestResource.php (hunk 2 above)
Comment (rodrigoaguilera) [reply modules/anu_lms_assessments/src/Plugin/rest/resource/QuestionRestResource.php:250]: Please link to the patch in the comment.

**Comment by rodrigoaguilera** [comment]
Right, the fix is only needed in one place

**Review by rodrigoaguilera — APPROVED** [review]
